### PR TITLE
Parentheses

### PR DESCRIPTION
**Pull request by galvisma**
Given a string s containing just the characters '(', ')', '{', '}', '[' and ']', determine if the input string is valid.
An input string is valid if:
Open brackets must be closed by the same type of brackets.
Open brackets must be closed in the correct order.
Every close bracket has a corresponding open bracket of the same type.

### CODE REVIEW

File: src/Parentheses/Main.java [hunk 1]
```diff
@@ -0,0 +1,13 @@
+package Parentheses;
+
+public class Main {
+    public static void main(String[] args) {
+
+        // String elements = "[}"
+        // String elements = "()";
```
Comment (julianespinel) [inline src/Parentheses/Main.java:7]: Please remove commented code.

File: src/Parentheses/Solution.java [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+import java.util.Stack;
+
+public class Solution {
+
+    public boolean isValid(String parentheses) {
+
+        char[] arrayParentheses = parentheses.toCharArray();            // O(parentheses)
+        int allOk = 0;                                                  // O(1)
+                                                                       // time          Space
+        Stack<Character> pila = new Stack<>(); // O(1)
```
Comment (julianespinel) [inline src/Parentheses/Solution.java:12]: Pila is a spanish word. Please translate it to English.

File: src/Parentheses/Solution.java [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+        }
+        if (pila.isEmpty() && allOk == 0) {                                     // O(1)     O(0)
+            return true;
+        } else {
+            return false;
+        }
+    }
+
+    private void closeParentheses(Stack<Character> pila) {
+    }
```
Comment (julianespinel) [inline src/Parentheses/Solution.java:73]: This method is empty. Please remove it if we are not using it.

File: src/Parentheses/Solution.java [hunk 4]
```diff
@@ -0,0 +1,74 @@
+package Parentheses;
+
+import java.util.Stack;
+
+public class Solution {
+
+    public boolean isValid(String parentheses) {
+
+        char[] arrayParentheses = parentheses.toCharArray();            // O(parentheses)
+        int allOk = 0;                                                  // O(1)
```
Comment (julianespinel) [inline src/Parentheses/Solution.java:10]: Could `allOk` be a `boolean` instead of an `int`?

File: src/Parentheses/Solution.java [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+                    default:
+                        throw new IllegalStateException("Unexpected value: " + arrayParentheses[i]);
+                }
+            }
+        }
+        if (pila.isEmpty() && allOk == 0) {                                     // O(1)     O(0)
+            return true;
+        } else {
+            return false;
+        }
```
Comment (julianespinel) [inline src/Parentheses/Solution.java:69]: You could replace that code fragment for the following line:
```java
return pila.isEmpty() && allOk == 0
```

File: src/Parentheses/Solution.java [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+                    case ']':
+                        if (pila.isEmpty()) {                                   // O(1)     O(0)
+                            allOk = 1;
+                        } else {
+                            char openCharacter = '[';
+                            if (pila.pop() != openCharacter) {                  // O(1)     O(1)
+                                allOk = 1;
+                            }
+                            System.out.println("pila: " + i + " " + pila);
+                        }
```
Comment (julianespinel) [inline src/Parentheses/Solution.java:58]: It looks like the body of each case could be extracted into a method. Then we could just call that method with different arguments. For example:
```java
checkNextParenthesis(pila, openCharacter)
```

**Review by julianespinel — COMMENTED** [review]
Great work! :clap: Just a few comments to take into account.

**Comment by galvisma** [comment]
This PR is closed in favor of Factored Parentheses, Evaluation method was created

**Pull request closed without merging**